### ISSUE TITLE
\DateTime is not implementing \DateTimeInterface (in php5.4)

### ISSUE THREAD

**Issue by CZechBoY** [kalessil/phpinspectionsea#357]
this code
```php
function ($date)
{
    if ($date instanceof \DateTime || $date instanceof \DateTimeInterface) {}
}
```
generates `This condition is ambiguous and can be safely removed.` (on first part of condition).
I know that it is because on PHP 5.5+ \DateTime implements \DateTimeInterface, but I have set php5.4 in my project settings. So this is nearly dangerous not false positive.

**Comment by kalessil** [comment]
Thank you for reporting @CZechBoY, uh this report is not going to be easy:
- please report to JB (https://youtrack.jetbrains.com/issues/WI, php stubs causing the issue), also please post here cross-link;
- I'll implement a workaround on our side to secure our community
